crmsh/sbd.py

```diff
@@ -3,14 +3,263 @@
+        """
+        Get the value for SBD_DELAY_START, formulas are:
+
+        SBD_DELAY_START = (token + consensus + pcmk_delay_max + msgwait)  # for disk-based sbd
+        SBD_DELAY_START = (token + consensus + 2*SBD_WATCHDOG_TIMEOUT) # for disk-less sbd
+        """
+        value_from_calculation = 0
+        token_consensus_margin = corosync.token_consensus_margin()
+        if self.disk_based:
+            value_from_calculation = token_consensus_margin + self.pcmk_delay_max + self.msgwait
+        else:
+            value_from_calculation = token_consensus_margin + 2*self.sbd_watchdog_timeout
+
+        value_from_configuration = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
+        if re.search(r'\d+', value_from_configuration):
+            adjust_value = max(int(value_from_configuration), value_from_calculation)
+        else:
+            adjust_value = value_from_calculation
+        return adjust_value
+
+    @staticmethod
+    def get_suitable_sbd_systemd_timeout():
+        """
+        Get suitable systemd start timeout for sbd.service
+        """
+        return int(1.2 * int(SBDManager.get_sbd_value_from_config("SBD_DELAY_START")))
+
+    @staticmethod
+    def is_delay_start():
+        """
+        Check if SBD_DELAY_START is not no or not set
+        """
+        res = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
+        return res and res != "no"
+
+    def adjust_systemd_start_timeout(self):
+        """
+        Adjust start timeout for sbd when set SBD_DELAY_START
+        """
+        if not self.two_node_without_qdevice and os.path.isdir(SBD_SYSTEMD_DELAY_START_DIR):
+            shutil.rmtree(SBD_SYSTEMD_DELAY_START_DIR)
+            bootstrap.csync2_update(SBD_SYSTEMD_DELAY_START_DIR)
+            utils.cluster_run_cmd("systemctl daemon-reload")
+            return
+
+        # TimeoutStartUSec default is 1min 30s, need to parse as seconds
+        cmd = "systemctl show -p TimeoutStartUSec sbd --value"
+        out = utils.get_stdout_or_raise_error(cmd)
+        res_seconds = re.search("(\d+)s", out)
+        default_start_timeout = int(res_seconds.group(1)) if res_seconds else 0
+        res_min = re.search("(\d+)min", out)
+        default_start_timeout += 60 * int(res_min.group(1)) if res_min else 0
+        if default_start_timeout >= self.sbd_delay_start_value:
+            return
+
+        utils.mkdirp(SBD_SYSTEMD_DELAY_START_DIR)
+        sbd_delay_start_file = "{}/sbd_delay_start.conf".format(SBD_SYSTEMD_DELAY_START_DIR)
+        utils.str2file("[Service]\nTimeoutSec={}".format(int(1.2*self.sbd_delay_start_value)), sbd_delay_start_file)
+        bootstrap.csync2_update(SBD_SYSTEMD_DELAY_START_DIR)
+        utils.cluster_run_cmd("systemctl daemon-reload")
+
+    def adjust_stonith_timeout(self):
+        """
+        Adjust stonith-timeout property
+        """
+        utils.set_property(stonith_timeout=self.get_stonith_timeout())
+
+    def adjust_pcmk_delay_max(self):
```

Comment:
TODO is needed, maybe in the future, and outside of sbd.py. This function applies to any fence device, not just sbd only.

Reply:
Already marked with TODO